src/mmw/js/src/modeling/models.js

```diff
@@ -221,6 +221,54 @@ var ScenariosCollection = Backbone.Collection.extend({
             }
             model.set('active', active);
         });
+    },
+
+    duplicateScenario: function(cid) {
+        var source = this.get(cid),
+            sourceMods = source.get('modifications').models,
+            newModel = new ScenarioModel({
+                is_current_conditions: false,
+                name: this.makeNewScenarioName('Copy of ' + source.get('name'))
+            });
+
+        newModel.set('modifications', new ModificationsCollection(sourceMods, {
+            model: ModificationModel
+        }));
```

Comment:
Unfortunately, because of some side-effect of nested models/collections not being fully supported, the collection is reset on initialization and the modifications are lost. I think if each model in `sourceMods` were being initialized as new models, we would be okay, but I'm not totally sure. Anyway, this was the simple work around. I'll remove the second argument and add a comment.